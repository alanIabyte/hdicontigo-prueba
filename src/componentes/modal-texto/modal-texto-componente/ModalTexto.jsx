import React, { useEffect } from "react";
import PropTypes from "prop-types";
import IconoCerrar from "@material-ui/icons/CloseRounded";
import {
  EnvolvedorModalTexto,
  ModalContenedor,
  EncabezadoModal,
  CuerpoModal,
  ContenedorIconoCierre,
  PiePaginaModal,
  LigaDescargaModal,
} from "./ModalTexto.styled";
import Boton from "../../boton";

const ModalTexto = (props) => {
  const {
    mostrar,
    manejarCierre,
    titulo,
    texto,
    archivo,
    nombreArchivo,
    textoBoton,
    runActionLog,
    manejarCierreFun,
  } = props;

  useEffect(() => {
    if (mostrar) {
      document.body.style.overflow = "hidden";
    } else {
      document.body.style.overflow = "unset";
    }
  }, [mostrar]);

  const despliegaTextoCuerpo = () => ({ __html: texto });

  return (
    <EnvolvedorModalTexto mostrar={mostrar}>
      <ModalContenedor>
        <EncabezadoModal>
          {titulo}
          <ContenedorIconoCierre
            onClick={() => {
              // eslint-disable-next-line no-unused-expressions
              !manejarCierreFun ? manejarCierre(false) : manejarCierre();
            }}
          >
            <IconoCerrar />
          </ContenedorIconoCierre>
        </EncabezadoModal>
        <CuerpoModal dangerouslySetInnerHTML={despliegaTextoCuerpo()} />
        {archivo && nombreArchivo && (
          <PiePaginaModal>
            <LigaDescargaModal
              target="_blank"
              rel="noreferrer"
              href={archivo}
              download={nombreArchivo}
            >
              <Boton etiqueta={textoBoton} enClick={runActionLog} />
            </LigaDescargaModal>
          </PiePaginaModal>
        )}
      </ModalContenedor>
    </EnvolvedorModalTexto>
  );
};

ModalTexto.propTypes = {
  titulo: PropTypes.string,
  texto: PropTypes.string,
  mostrar: PropTypes.bool,
  manejarCierre: PropTypes.func,
  textoBoton: PropTypes.string,
  archivo: PropTypes.string,
  nombreArchivo: PropTypes.string,
  runActionLog: PropTypes.func,
  manejarCierreFun: PropTypes.bool,
};

ModalTexto.defaultProps = {
  titulo: "",
  texto: "",
  mostrar: false,
  manejarCierre() {},
  textoBoton: "",
  archivo: "",
  nombreArchivo: "",
  runActionLog: () => {},
  manejarCierreFun: false,
};

export default ModalTexto;
