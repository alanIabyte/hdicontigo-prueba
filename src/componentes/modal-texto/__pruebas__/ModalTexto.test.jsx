import React from "react";
import renderer from "react-test-renderer";
import ModalTexto from "../modal-texto-componente/ModalTexto";

describe("#ModalTexto", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<ModalTexto />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
