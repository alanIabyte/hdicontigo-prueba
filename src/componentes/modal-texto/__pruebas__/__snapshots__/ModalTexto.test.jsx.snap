// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ModalTexto se despliega correctamente 1`] = `
<div
  className="sc-ikkxIA jHOaZv"
>
  <div
    className="sc-dAbbOL GffIY"
  >
    <div
      className="sc-fqkvVR sc-feUZmu wQLUp tfGny"
    >
      
      <button
        className="sc-hzhJZQ blnAos"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.3 5.71a.9959.9959 0 00-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
          />
        </svg>
      </button>
    </div>
    <div
      className="sc-iGgWBj sc-fUnMCh fgQoGq gpwtAM"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    
  </div>
</div>
`;
